### PR TITLE
Add Guest and Hugepages memory to kubevirt_vm_resource_requests

### PR DESCRIPTION
**Pull request by machadovilaca**
### What this PR does
Before this PR:

Guest and Hugepages memory requests were not included in the kubevirt_vm_resource_requests metric

After this PR:

The metric also exposes these values

jira-ticket: `https://issues.redhat.com/browse/CNV-51408`

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
Add Guest and Hugepages memory to kubevirt_vm_resource_requests
```

### CODE REVIEW

**Comment by sradco** [comment]
@machadovilaca will this allow us to get the allocated memory for every VM , both running and non-running?
We need to have this even if a request was not set specifically.
It also needs to be the exact number that the User requested.
So for example, If the user asked for 2GB memory and we are creating the VM with 2.15GB, the metric should report 2GB.

**Comment by machadovilaca** [comment]
> @machadovilaca will this allow us to get the allocated memory for every VM , both running and non-running? We need to have this even if a request was not set specifically. It also needs to be the exact number that the User requested. So for example, If the user asked for 2GB memory and we are creating the VM with 2.15GB, the metric should report 2GB.

It is the exact memory requested in the VM spec. And the user must always set one of these values, Resources.Requests, Memory.Guest or Memory.Hugepages

File: pkg/monitoring/metrics/virt-controller/vmstats_collector.go [hunk 1]
```diff
@@ -122,7 +122,7 @@ var (
 			Name: "kubevirt_vm_resource_requests",
 			Help: "Resources requested by Virtual Machine. Reports memory and CPU requests.",
```
Comment (avlitman) [inline pkg/monitoring/metrics/virt-controller/vmstats_collector.go:123]: can we update the description with the new label? 
and a question for me - the source can be domain/gust/hugepages? is there more options?

Comment (machadovilaca) [reply pkg/monitoring/metrics/virt-controller/vmstats_collector.go:123]: I don't think so

**Pull request merged**